timelock-agent/src/main/java/com/palantir/timelock/paxos/AsyncTimeLockServicesCreator.java

```diff
@@ -73,16 +74,45 @@ public TimeLockServices createTimeLockServices(
                         : JavaSuppliers.compose(NonTransactionalLockService::new, rawLockServiceSupplier),
                 client);
 
+        if (runtime.get().clientTokens().containsKey(client)) {
```

Comment:
when does this execute?  Does it happen with every call or just at startup?

Reply:
This is just at startup. We could make it happen at every call by passing the runtime config into the `SecureTimelockResource` (or `SecureTimelockService`) - the resource would then need to handle both `v1` (secured) calls and unsecured calls.